File: backend/app/domain/enums.py
from enum import Enum


class CapabilityCategory(str, Enum):
    BACKEND = "backend"
    CLOUD = "cloud"
    AI = "ai"
    DATA = "data"
    DEVOPS = "devops"
    ARCHITECTURE = "architecture"
    SECURITY = "security"
    DELIVERY_EXECUTION = "delivery_execution"


class CoverageLevel(str, Enum):
    MISSING = "missing"
    WEAK = "weak"
    ADEQUATE = "adequate"
    STRONG = "strong"


class ConfidenceLevel(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class RiskSeverity(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class RiskFindingType(str, Enum):
    MISSING_CRITICAL_CAPABILITY = "missing_critical_capability"
    WEAK_CAPABILITY = "weak_capability"
    KEY_PERSON_DEPENDENCY = "key_person_dependency"
    INCOMPLETE_EVIDENCE = "incomplete_evidence"
    DUPLICATE_TEAM_MEMBER = "duplicate_team_member"
    EMPTY_TEAM = "empty_team"


class ReadinessDimension(str, Enum):
    CAPABILITY_COVERAGE = "capability_coverage"
    SKILL_DEPTH = "skill_depth"
    TEAM_BALANCE = "team_balance"
    DELIVERY_RISK = "delivery_risk"
    EVIDENCE_QUALITY = "evidence_quality"


class EvidenceSource(str, Enum):
    SKILLS = "skills"
    CERTIFICATIONS = "certifications"
    PROJECTS = "projects"
    EXPERIENCE = "experience"


class SimulationOperationType(str, Enum):
    ADD = "add"
    REMOVE = "remove"
    REPLACE = "replace"
    COMPARE = "compare"


class SimulationChangeType(str, Enum):
    INTRODUCED = "introduced"
    RESOLVED = "resolved"
    ESCALATED = "escalated"
    DEESCALATED = "deescalated"
    IMPROVED = "improved"
    DEGRADED = "degraded"
    MODIFIED = "modified"


class MitigationType(str, Enum):
    ADD_CAPABILITY_COVERAGE = "add_capability_coverage"
    STRENGTHEN_CAPABILITY_COVERAGE = "strengthen_capability_coverage"
    ESTABLISH_SECONDARY_OWNER = "establish_secondary_owner"
    PRESERVE_CRITICAL_ENGINEER = "preserve_critical_engineer"
    IMPROVE_ENGINEER_EVIDENCE = "improve_engineer_evidence"
    REASSESS_PROJECT_SCOPE = "reassess_project_scope"
    REPLACE_WITH_STRONGER_MATCH = "replace_with_stronger_match"


class MitigationPriority(str, Enum):
    CRITICAL = "critical"
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"


class HumanReviewState(str, Enum):
    ACCEPTED = "accepted"
    OVERRIDDEN = "overridden"
    NEEDS_MORE_DATA = "needs_more_data"


class AuditEventType(str, Enum):
    ASSESSMENT_CREATED = "assessment_created"
    SIMULATION_CREATED = "simulation_created"
    HUMAN_REVIEW_CREATED = "human_review_created"
    LEADERSHIP_BRIEF_CREATED = "leadership_brief_created"


class AuditAggregateType(str, Enum):
    ASSESSMENT = "assessment"
    SIMULATION = "simulation"
    HUMAN_REVIEW = "human_review"


class ScenarioType(str, Enum):
    READINESS = "readiness"
    SIMULATION = "simulation"

File: frontend/src/lib/api/contracts/enums.ts
export type CapabilityCategory =
  | "backend"
  | "cloud"
  | "ai"
  | "data"
  | "devops"
  | "architecture"
  | "security"
  | "delivery_execution";

export type CoverageLevel = "missing" | "weak" | "adequate" | "strong";

export type ConfidenceLevel = "low" | "medium" | "high";

export type RiskSeverity = "low" | "medium" | "high";

export type RiskFindingType =
  | "missing_critical_capability"
  | "weak_capability"
  | "key_person_dependency"
  | "incomplete_evidence"
  | "duplicate_team_member"
  | "empty_team";

export type ReadinessDimension =
  | "capability_coverage"
  | "skill_depth"
  | "team_balance"
  | "delivery_risk"
  | "evidence_quality";

export type EvidenceSource =
  | "skills"
  | "certifications"
  | "projects"
  | "experience";

export type SimulationOperationType = "add" | "remove" | "replace" | "compare";

export type SimulationChangeType =
  | "introduced"
  | "resolved"
  | "escalated"
  | "deescalated"
  | "improved"
  | "degraded"
  | "modified";

export type MitigationType =
  | "add_capability_coverage"
  | "strengthen_capability_coverage"
  | "establish_secondary_owner"
  | "preserve_critical_engineer"
  | "improve_engineer_evidence"
  | "reassess_project_scope"
  | "replace_with_stronger_match";

export type MitigationPriority = "critical" | "high" | "medium" | "low";

export type HumanReviewState = "accepted" | "overridden" | "needs_more_data";

export type LeadershipDecision =
  | "proceed"
  | "proceed_with_conditions"
  | "defer"
  | "do_not_proceed";

export type ProviderMode = "azure_openai" | "deterministic_fallback";

export type GenerationStatus =
  | "generated"
  | "fallback_generated"
  | "failed";

export type LeadershipBriefFailureCategory =
  | "ai_disabled"
  | "missing_configuration"
  | "timeout"
  | "authentication_error"
  | "rate_limited"
  | "provider_unavailable"
  | "malformed_output"
  | "schema_validation_failed"
  | "grounding_validation_failed"
  | "empty_output"
  | "unknown_provider_error";

export type LeadershipActionPriority = "critical" | "high" | "medium" | "low";

export type LeadershipBriefRiskSeverity = "low" | "medium" | "high";
